test/e2e/scenarios/retina/retina_scenarios_test.go

```diff
@@ -7,80 +7,44 @@ import (
-	"github.com/microsoft/retina/test/e2e/framework/generic"
-	"github.com/microsoft/retina/test/e2e/framework/kubernetes"
+	"github.com/microsoft/retina/test/e2e/common"
 	"github.com/microsoft/retina/test/e2e/framework/types"
-	"github.com/microsoft/retina/test/e2e/scenarios/retina/drop"
-	tcp "github.com/microsoft/retina/test/e2e/scenarios/retina/tcp"
+	jobs "github.com/microsoft/retina/test/e2e/jobs"
+	"github.com/stretchr/testify/require"
 )
 
-const (
-	// netObsRGtag is used to tag resources created by this test suite
-	netObsRGtag = "-e2e-netobs-"
-)
+// TestE2ERetina tests all e2e scenarios for retina
+func TestE2ERetina(t *testing.T) {
+	curuser, err := user.Current()
+	require.NoError(t, err)
 
-// Test against AKS cluster with NPM enabled,
-// create a pod with a deny all network policy and validate
-// that the drop metrics are present in the prometheus endpoint
-func TestE2ERetinaMetrics(t *testing.T) {
-	job := types.NewJob("Validate that drop metrics are present in the prometheus endpoint")
-	runner := types.NewRunner(t, job)
-	defer runner.Run()
+	clusterName := curuser.Username + common.NetObsRGtag + strconv.FormatInt(time.Now().Unix(), 10)
 
-	curuser, _ := user.Current()
+	subID := os.Getenv("AZURE_SUBSCRIPTION_ID")
+	require.NotEmpty(t, subID)
 
-	testName := curuser.Username + netObsRGtag + strconv.FormatInt(time.Now().Unix(), 10)
-	sub := os.Getenv("AZURE_SUBSCRIPTION_ID")
-	loc := os.Getenv("AZURE_LOCATION")
-	if loc == "" {
-		loc = "eastus"
+	location := os.Getenv("AZURE_LOCATION")
+	if location == "" {
+		location = "eastus"
 	}
-
-	job.AddStep(&azure.CreateResourceGroup{
-		SubscriptionID:    sub,
-		ResourceGroupName: testName,
-		Location:          loc,
-	}, nil)
-
-	job.AddStep(&azure.CreateVNet{
-		VnetName:         "testvnet",
-		VnetAddressSpace: "10.0.0.0/9",
-	}, nil)
-
-	job.AddStep(&azure.CreateSubnet{
-		SubnetName:         "testsubnet",
-		SubnetAddressSpace: "10.0.0.0/12",
-	}, nil)
-
-	job.AddStep(&azure.CreateNPMCluster{
-		ClusterName:  testName,
-		PodCidr:      "10.128.0.0/9",
-		DNSServiceIP: "192.168.0.10",
-		ServiceCidr:  "192.168.0.0/28",
-	}, nil)
-
-	job.AddStep(&azure.GetAKSKubeConfig{
-		KubeConfigFilePath: "./test.pem",
-	}, nil)
-
-	job.AddStep(&generic.LoadFlags{
-		TagEnv:            generic.DefaultTagEnv,
-		ImageNamespaceEnv: generic.DefaultImageNamespace,
-		ImageRegistryEnv:  generic.DefaultImageRegistry,
-	}, nil)
-
-	// todo: enable mutating images in helm chart
-	job.AddStep(&kubernetes.InstallHelmChart{
-		Namespace:   "kube-system",
-		ReleaseName: "retina",
-		ChartPath:   "../../../../deploy/manifests/controller/helm/retina/",
-	}, nil)
-
-	job.AddScenario(drop.ValidateDropMetric())
-
-	// todo: handle multiple scenarios back to back
-	job.AddScenario(tcp.ValidateTCPMetrics())
-
-	job.AddStep(&azure.DeleteResourceGroup{}, nil)
+	// CreateTestInfra
+	createTestInfra := types.NewRunner(t, jobs.CreateTestInfra(subID, clusterName, location))
```

Comment:
Interesting take, my original thesis was:

- Step is code, reusable, string parameters like a CLI command for easy prevalidation. No dependency on go.test
- Scenario is a slice of steps. No dependency on go.test
- Job is a logical sequence of scenarios, and parameters between each scenario's steps can be shared. (Ex: if I create a cluster, validation can happen before anything is created that tests that need cluster, will have cluster by the time they're executed)
- Runner wraps single job, and _is_ go.test dependent

If we break things into separate runners, then tests might not have extra context about the environment that they're running on. No right way or wrong way, but the thesis of coupling infra deployment + tests was that infra info could be set once, and tests wouldn't care about fetching the right kubeconfig, since they'd be running in the context of the infra created with it.

This seems to accomplish that task, by passing arguments to jobs. Will read more for a full review